Source/Tools/FEXServer/ProcessPipe.cpp

```diff
@@ -448,92 +383,66 @@ namespace ProcessPipe {
     close(ServerSocketFD);
   }
 
-  void WaitForRequests() {
-    auto LastDataTime = std::chrono::system_clock::now();
-
-    while (!ShouldShutdown) {
-      struct timespec ts{};
-      ts.tv_sec = RequestTimeout;
-
-      int Result = ppoll(&PollFDs.at(0), PollFDs.size(), &ts, nullptr);
-      std::vector<struct pollfd> NewPollFDs{};
-
-      if (Result > 0) {
-        // Walk the FDs and see if we got any results
-        for (auto it = PollFDs.begin(); it != PollFDs.end(); ) {
-          auto &Event = *it;
-          bool Erase{};
-
-          if (Event.revents != 0) {
-            if (Event.fd == ServerSocketFD) {
-              if (Event.revents & POLLIN) {
-                // If it is the listen socket then we have a new connection
-                struct sockaddr_storage Addr{};
-                socklen_t AddrSize{};
-                int NewFD = accept(ServerSocketFD, reinterpret_cast<struct sockaddr*>(&Addr), &AddrSize);
-
-                // Add the new client to the temporary array
-                NewPollFDs.emplace_back(pollfd {
-                  .fd = NewFD,
-                  .events = POLLIN | POLLPRI | POLLRDHUP,
-                  .revents = 0,
-                });
-              }
-              else if (Event.revents & (POLLHUP | POLLERR | POLLNVAL)) {
-                // Listen socket error or shutting down
-                break;
-              }
-            }
-            else {
-              if (Event.revents & POLLIN) {
-                // Data from the socket
-                HandleSocketData(Event.fd);
-              }
-
-              if (Event.revents & (POLLHUP | POLLERR | POLLNVAL | POLLRDHUP)) {
-                // Error or hangup, close the socket and erase it from our list
-                Erase = true;
-                close(Event.fd);
-              }
-            }
-
-            Event.revents = 0;
-            --Result;
-          }
+  class ProcessPipeHandler final : public FEXServer::SocketConnectionHandler {
```

Comment:
Either this must use non-`public` inheritance or `SocketConnectionHandler` must have a virtual destructor. (Currently, the code `SocketConnectionHandler* x = new ProcessPipeHandler(...); delete x;` would cause a resource leak.)